CONTRIBUTING.md

```diff
@@ -78,6 +78,52 @@ make lint
 
 Note: The lint tasks require the [codespell](https://github.com/codespell-project/codespell) application. Be sure to install version 2.0 or newer before running the lint tasks.
 
+### Local Development/Testing
+
+It's very helpful to be able to test out changes to Contour locally without building images and pushing into clusters.
+
+To accomplish this, Envoy can be run inside a Kubernetes cluster, typically a `kind` cluster.
+Then Contour is run on your local machine and Envoy will be configured to look for Contour running on your machine vs running in the cluster.
+
+1. Create a kind cluster
+
+```shell
+kind create cluster --config=./examples/kind/kind-expose-port.yaml --name=contour
+```
+
+2. Deploy Contour & Deps to cluster:
+
+```shell
+kubectl apply -f examples/contour
+```
+
+3. Find IP of local machine (e.g. `ifconfig` or similar depending on your environment)
+
+4. Edit Envoy Daemonset & change the xds-server value to your local IP address
+```shell
+kubectl edit ds envoy -n projectcontour
+```
+
+Change `initContainers:` to look like this updating the IP and removing the three envoy cert flags:
+```shell
+ initContainers:
+  - args:
+    - bootstrap
+    - /config/envoy.json
+    - --xds-address=<YOUR_IP_ADDRESS>
+    - --xds-port=8001
+    - --xds-resource-version=v3
+    - --resources-dir=/config/resources
+```
+
+5. Change your Contour code.
+
+6. Test using the local kind cluster by deploying resources into that cluster. Envoy will connect to your local Contour instance and get it's updated configuration allowing for local testing.
+```shell
+make install && contour serve --kubeconfig=$HOME/.kube/config --xds-address=0.0.0.0 --insecure 
+```
```

Comment:
I'd split this into two steps, the first being just "Start Contour locally" with the command you have, and then the next being "Test using the local kind cluster by ..."

Reply:
Yeah, I agree, more, simpler steps are better in this case.